ruby/red-arrow/lib/arrow/tensor.rb

```diff
@@ -15,8 +15,140 @@
+    #     bytes in each dimension.
+    #
+    #     If you specify `nil` or an empty `Array`, strides are
+    #     guessed from `raw_tensor`.
+    #
+    #   @param dimension_names [::Array<String>, ::Array<Symbol>, nil]
+    #     The array of the dimension names.
+    #
+    #     If you specify `nil`, all dimensions doesn't have their names.
+    #
+    #   @example Create a tensor from Ruby's Array
+    #     raw_tensor = [
+    #       [
+    #         [1, 2, 3, 4],
+    #         [5, 6, 7, 8],
+    #       ],
+    #       [
+    #         [9, 10, 11, 12],
+    #         [13, 14, 15, 16],
+    #       ],
+    #       [
+    #         [17, 18, 19, 20],
+    #         [21, 22, 23, 24],
+    #       ],
+    #     ]
+    #     Arrow::Tensor.new(raw_tensor)
+    #
+    #   @since 10.0.0
+    #
+    # @overload initialize(data_type, data, shape, strides, dimension_names)
+    #
+    #   @param data_type [Arrow::DataType, String, Symbol, ::Array<String>,
+    #     ::Array<Symbol>, Hash] The element data type of the tensor.
+    #
+    #     See {Arrow::DataType.resolve} how to specify data type.
+    #
+    #   @param data [Arrow::Buffer] The data of the tensor.
+    #
+    #   @param shape [::Array<Integer>] The array of dimension sizes.
+    #
+    #   @param strides [::Array<Integer>, nil] The array of the number of
+    #     bytes in each dimension.
+    #
+    #     If you specify `nil` or an empty `Array`, strides are
+    #     guessed from `data_type` and `data`.
+    #
+    #   @param dimension_names [::Array<String>, ::Array<Symbol>, nil]
+    #     The array of the dimension names.
+    #
+    #     If you specify `nil`, all dimensions doesn't have their names.
+    #
+    #   @example Create a table from Arrow::Buffer
+    #     raw_data = [
+    #       1, 2,
+    #       3, 4,
+    #
+    #       5, 6,
+    #       7, 8,
+    #
+    #       9, 10,
+    #       11, 12,
+    #     ]
+    #     data = Arrow::Buffer.new(raw_data.pack("c*").freeze)
+    #     shape = [3, 2, 2]
+    #     strides = []
+    #     names = ["a", "b", "c"]
+    #     Arrow::Tensor.new(:int8, data, shape, strides, names)
+    def initialize(*args,
+                   data_type: nil,
+                   data: nil,
+                   shape: nil,
+                   strides: nil,
+                   dimension_names: nil)
+      n_args = args.size
+      case n_args
+      when 1
+        converter = RawTensorConverter.new(args[0],
+                                           data_type: data_type,
+                                           shape: shape,
+                                           strides: strides,
+                                           dimension_names: dimension_names)
+        data_type = converter.data_type
+        data = converter.data
+        shape = converter.shape
+        strides = converter.strides
+        dimension_names = converter.dimension_names
+      when 5
```

Comment:
Is it common to allow this in Ruby?

Reply:
What does the "this" mean? Dispatching by the number of positional arguments?

Reply:
Yes :-) Is there no easier way to do this?

Reply:
There is no easier way for this in Ruby. :(
Because Ruby doesn't support method overload. If we want to define multiple signature for the same name method, we need to dispatch by arguments in the method. If the number of arguments are different for each signature, we can use it to dispatch. If the number of arguments are same, we can use `args[0].is_a?(String)`, `args[0].respond_to?(:write)` or something.